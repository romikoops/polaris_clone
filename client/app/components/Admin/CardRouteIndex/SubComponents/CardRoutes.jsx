import React, { Component } from 'react'
import { withNamespaces } from 'react-i18next'
import PropTypes from 'prop-types'
import styles from '../Card.scss'
import { gradientTextGenerator, switchIcon } from '../../../../helpers'

class CardRoutes extends Component {
  constructor (props) {
    super(props)
    this.state = {}
    this.selectItinerary = this.selectItinerary.bind(this)
  }

  selectItinerary () {
    const { itinerary, handleClick } = this.props
    handleClick(itinerary)
  }

  render () {
    const {
      handleClick, onDisabledClick, disabled, itinerary, theme, t
    } = this.props
    const gradientFontStyle =
      theme && theme.colors
        ? gradientTextGenerator(theme.colors.primary, theme.colors.secondary)
        : { color: '#E0E0E0' }
    const disabledClass = disabled ? styles.disabled : ''
    const [originNexus, destinationNexus] = itinerary.name.split(' - ')

    return (
      <div
        className="card_padding flex-100 flex-md-50 flex-gt-md-33"
        onClick={disabled ? onDisabledClick : () => handleClick(itinerary.id)}
      >
        <div
          className={`${styles.card_route_pricing} ${disabledClass} margin_bottom flex`}

        >
          <div className={styles.top_routes}>
            <div>
              <p>
                {t('common:from')}:{' '}
                <strong>
                  <span> {originNexus} </span>
                </strong>
              </p>
              <p>
                {t('common:to')}:{' '}
                <strong>
                  <span> {destinationNexus} </span>
                </strong>
              </p>
            </div>
            {switchIcon(itinerary.mode_of_transport, gradientFontStyle)}
          </div>
        </div>
      </div>
    )
  }
}

CardRoutes.propTypes = {
  t: PropTypes.func.isRequired,
  itinerary: PropTypes.objectOf(PropTypes.any).isRequired,
  handleClick: PropTypes.func.isRequired,
  onDisabledClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  theme: PropTypes.theme
}
CardRoutes.defaultProps = {
  disabled: false,
  theme: {}
}

export default withNamespaces('common')(CardRoutes)
