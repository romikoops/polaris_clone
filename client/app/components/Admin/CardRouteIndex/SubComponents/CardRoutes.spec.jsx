import * as React from 'react'
import { shallow } from 'enzyme'

import { theme, identity } from '../../../../mock'
import CardRoutes from './CardRoutes'

const propsBase = {
  theme,
  itinerary: { name: 'FOO - BAR' },
  handleClick: identity,
  onDisabledClick: identity,
  disabled: false
}

test('shallow render', () => {
  expect(shallow(<CardRoutes {...propsBase} />)).toMatchSnapshot()
})
