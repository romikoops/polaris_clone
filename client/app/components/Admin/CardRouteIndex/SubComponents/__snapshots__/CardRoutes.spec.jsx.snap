// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow render 1`] = `
<div
  className="card_padding flex-100 flex-md-50 flex-gt-md-33"
  onClick={[Function]}
>
  <div
    className="card_route_pricing  margin_bottom flex"
  >
    <div
      className="top_routes"
    >
      <div>
        <p>
          From
          :
           
          <strong>
            <span>
               
              FOO
               
            </span>
          </strong>
        </p>
        <p>
          To
          :
           
          <strong>
            <span>
               
              BAR
               
            </span>
          </strong>
        </p>
      </div>
      <i
        className="fa fa-anchor clip "
        style={
          Object {
            "background": "-webkit-linear-gradient(-90deg, #333,#fafafa)",
          }
        }
      />
    </div>
  </div>
</div>
`;
